### ISSUE TITLE
antiaim

### ISSUE THREAD

**Issue by ogosirisfan** [ijre/Osiris#6]
when the knife is anti aim does not work I would like the inscription to be red when the knife, how to do?

**Comment by ijre** [comment]
Yes, the antiaim turns off when you have certain weapons equiped, though it could use some tweaking so that it only turns off when you attack with those certain weapons.

When you say "the inscription", do you mean the text that says your AA settings?

**Comment by ogosirisfan** [comment]
Yes

**Comment by ijre** [comment]
Right now I'm implementing a fix that will make it so that knifes and such don't disable AA, and only turn off for a quick moment when attacking like normal weapons do; would you still want the red text change?

**Comment by ogosirisfan** [comment]
if you change that the knife has antiaim then no, but when you automatically fire or press mouse1 this yes

**Comment by ijre** [comment]
Just to notify you, if I have it turn red when firing, it'll flicker quickly back and forth from your selected colour to red when automatically shooting, and more than likely look like shit, you sure you still want it?

Also, here's the fix:

Where it says "if (activeWeapon->isBannedFromWepMagESP())", at BOTH Antiaim::pitch and Antiaim::yaw, replace it with:

"if (activeWeapon->isBannedFromWepMagESP() && cmd->buttons & UserCmd::IN_ATTACK2)", so that it says:
```
if (activeWeapon->isBannedFromWepMagESP() && cmd->buttons & UserCmd::IN_ATTACK2)
    return;
```

Make sure that you do this for both functions as well

**Comment by ogosirisfan** [comment]
if you don't want don't add red text to your project but you can share the code here.
Do you have plans to add a resolver? focus more accurately on the rage bot?

**Comment by ijre** [comment]
Oh yeah I don't mind sharing the code for it here, I just wanted to make sure it was something you truly wanted is all

As for a resolver, if I did add one it would probably be pasted as I've never touched that sort of thing and have no idea what to look for to make it good, though it is something I want to add to at least marginally improve the cheat

And for the rage bot, I'm currently working on improving the bolt actions so that it waits for accuracy before shooting as well as jump scouting

Anyways, here's the code for the antiaim text:

```
		float red[3]{ 255.0f, 0.0f, 0.0f };
		float* colourArg = (firing ? red : config.antiaim.drawPitchDirectionColour);

		if (config.antiaim.pitchStatic)
			renderPositionedTextForAntiaim(L"STATIC", colourArg, { pitchFirstArg,  pitchSecondArg });

		if (config.antiaim.pitchDown)
			renderPositionedTextForAntiaim(L"DOWN", colourArg, { pitchFirstArg,  pitchSecondArg });

		if (config.antiaim.pitchUp)
			renderPositionedTextForAntiaim(L"UP", colourArg, { pitchFirstArg,  pitchSecondArg });

		if (config.antiaim.pitchJitter)
			renderPositionedTextForAntiaim(L"JITTER", colourArg, { pitchFirstArg, pitchSecondArg });



		if (config.antiaim.yawBackwards)
			renderPositionedTextForAntiaim(L"BACKWARDS", colourArg, { yawFirstArg,  yawSecondArg });

		if (config.antiaim.yawJitter)
			renderPositionedTextForAntiaim(L"JITTER", colourArg, { yawFirstArg,  yawSecondArg });

		if (config.antiaim.yawSpin)
			renderPositionedTextForAntiaim(L"SPINNING", colourArg, { yawFirstArg,  yawSecondArg });

		if (config.antiaim.yawSideways)
			renderPositionedTextForAntiaim(L"SIDEWAYS", colourArg, { yawFirstArg,  yawSecondArg });

		if (config.antiaim.yawLeft)
			renderPositionedTextForAntiaim(L"LEFT", colourArg, { yawFirstArg,  yawSecondArg });

		if (config.antiaim.yawRight)
			renderPositionedTextForAntiaim(L"RIGHT", colourArg, { yawFirstArg,  yawSecondArg });
```
In antiaim.h, inside the Antiaim namespace:
```bool firing;```
Don't worry about the missing "fake" settings, they're being removed as the method used was patched out a while ago, I'm going to be pushing that update after I upload this comment.

**Comment by ijre** [comment]
Oh, and inside antiaim::pitch and antiaim::yaw:
```
	if (cmd->buttons & UserCmd::IN_ATTACK) {
		firing = true;
		return;
	}
	else firing = false;
```

**Comment by ijre** [comment]
Move the "UserCmd::IN_USE" check to its own if statement which returns

**Comment by ijre** [comment]
Also, I haven't checked to make sure this works, so please let me know
